### DISCUSSION TITLE
How to feed BirdnetPi from an mp3 file rather than live acoustic

### DISCUSSION THREAD

**Discussion by ThickKnee** [mcguirepr89/BirdNET-Pi#473] [Q&A]
Hi.
We've been using BirdnetPi for a few months now and it's quite something. Using a Pi4 with 8Gb.

Sometimes we place a recorder in the bush for a number of hours. That produces .wav files. Back home, we play the files thru a laptop with a mic close to the speakers and the mic feeds into BirdnetPi on the RPi4 and BirdnetPi does the analysis. So we're recreating the bird calls acoustically. Cumbersome and vulnerable to ambient sounds.

Is there a way we can feed a .wav file directly into BirdnetPi without converting it to sound?

Many thanks to Patrick and all else who contributed to BirdnetPi.

John.

**Comment by ehpersonal38** [comment]
It's easy enough to drop your .wav files in `~/BirdSongs/`, where they'll get picked up and processed just like native recordings. The trickier bit will be getting the file name format of each .wav to match the expected one before you drop them in, something like: `2022-06-13-birdnet-10:31:44.wav`. You could probably make some python or shell script to do that by parsing the created date of the file.

Also I believe the model only accepts 48kHZ recordings. If your .wav's aren't already formatted this way, you can make the conversion with something like `ffmpeg -y -i recording.wav -ar 48000 recording_new.wav`

**Reply by ThickKnee** [reply]
Thank you. I'll try it out. The recorder says the sampling rate is 48k, so I suppose that's what you referred to.

The internal workings of Birdnet are largely a black box to me, but I think I know where the file has to be put. It will be about two hours in duration -- will that cause any difficulty? Will BirdnetPi still extract all bird calls inside that two hours?

**Reply by ehpersonal38** [reply]
I don't know the answer to that. You're certainly free to try it and if it doesn't work, you could split them up into several smaller files. Some quick maths tells me that if it does work, the analysis will take about 5-15 mins, so give it enough time to do its thing.

**Reply by ThickKnee** [reply]
I'm not having success. this is the folder where I thought the .wav file ought to go:
......./BirdSongs/July-2022/30-Saturday 

I see that BirdnetPi is continually creating .wav files in this folder every 12 seconds. So if I put my recorded file there it will get processed? Not easy to follow what's going on.

I made the file name:
2022-07-30-birdnet-13:45:09 (tricky to get the ":" into the file name)
The file gets removed but I see no indication that it was analysed.

Incidentally, the 12 sec .wav files in this folder keep getting created even when BirdnetPi is NOT running.

**Reply by ehpersonal38** [reply]
Make sure you have "View Log" open so you can see exactly what happens. I just tested your steps above and it got picked up and analyzed within a few seconds. Stop the "birdnet_recording" service if you don't want other files being created.
